### ISSUE TITLE
2.0.13 Cannot read property 'NONE' of undefined [Chrome/ window.Connection]

### ISSUE THREAD

**Issue by acramatte** [angular-ui/angular-google-maps#1186]
Hey there,

Since 2.0.13 I'm having issue on the [window.Connection.NONE](https://github.com/angular-ui/angular-google-maps/blob/master/dist/angular-google-maps.js#L107) used by the uiGmapMapScriptLoader factory

You can see it [on the demo page](http://angular-ui.github.io/angular-google-maps/#!/demo) with Chrome.

No issue with Firefox

**Comment by nmccready** [comment]
Whats the issue? I don't see any error on chrome. Please be more descriptive. This may be a fun bug as I did not add this feature.. :(

**Comment by nmccready** [comment]
I can probably add in some null saftey checks to avoid checking some of those things added for window.connection if it is missing.

**Comment by acramatte** [comment]
I cannot really be more descriptive than that:
```
TypeError: Cannot read property 'NONE' of undefined
    at Object.load (angular-google-maps_dev_mapped.js:107)
    at $get (angular-google-maps_dev_mapped.js:134)
    at Object.e [as invoke] (angular.min.js:36)
    at angular.min.js:38
    at d (angular.min.js:36)
    at Object.e [as invoke] (angular.min.js:36)
    at Object.$get (angular.min.js:34)
    at Object.e [as invoke] (angular.min.js:36)
    at angular.min.js:38
    at d (angular.min.js:36)
```

Strangely enough, it doesn't work on my Chrome 40.0.2214.115 (64-bit) but I don't have the issue on my Chrome Canary 43.0.2315.0 canary (64-bit). Maybe it's just me then

**Comment by nmccready** [comment]
Hmm. have you tried vanilla chrome without extensions?

**Comment by acramatte** [comment]
I was able to reproduce it on Chrome for Android 40.0.2214.109 and Beta 41.0.2272.63 on 2 different phones...

I don't get it. I'll do what you said and give a try with a vanilla Chrome.

**Comment by acramatte** [comment]
Ok sometimes reading the code makes it easier to understand :)

- Short version:

yes we need a null safety check on ```window.Connection```.

- Long version:

As written [here](https://github.com/angular-ui/angular-google-maps/commit/208812570f46d4fc828b3a147819f291cf04c0eb#diff-ea39d8646c25af66b445e997dd60772fR41) this window.Connection is Cordova specific. I have no idea how/when it gets added to the window object but whatever. 

 ```window.navigator.connection``` returns this on Canary Chrome 
```
undefined
```
which is, I guess, also what is returned for your Chrome.

This is what is returned by my Android Chrome:
```
{
  "ontypechange": null,
  "type": "cellular"
}
```
and for obscures and irrelevant reasons, this on my desktop Chrome:
```
{
  "ontypechange": null,
  "type": "unknown"
}
```
So basically every time  ```window.navigator.connection``` returns something with a type property then it complains because window.Connection doesn't exist (besides the Cordova world I guess).

**Comment by nmccready** [comment]
Ok, yeah I should have forced some specs on the Cordova guy who made the PR. Thanks for digging. If you want to but the null checks in feel free to modify the coffee file.

**Comment by QuentinFchx** [comment]
sorry :3
